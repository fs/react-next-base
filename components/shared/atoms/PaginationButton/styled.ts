import styled, { css, FlattenSimpleInterpolation } from 'styled-components';

interface IStyles {
  customStyles?: FlattenSimpleInterpolation | string;
}

export const StyledButton = styled.button<IStyles>(
  ({
    theme: {
      colors: { grey },
    },
    disabled,
    customStyles,
  }) => css`
    display: flex;
    padding: 0.5rem;
    background-color: transparent;
    border: 2px solid ${grey};

    ${disabled
      ? css`
          cursor: default;
          svg {
            fill: ${grey};
            transition: fill 0.2s;
          }
        `
      : css`
          & {
            background-color: transparent;
            transition: background-color 0.2s;
          }
          &:hover {
            background-color: rgba(0, 0, 0, 0.05);
            transition: background-color 0.2s;
          }
        `}

    ${customStyles}
  `,
);
