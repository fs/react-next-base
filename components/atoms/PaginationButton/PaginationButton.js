import React from 'react';
import styled, { css } from 'styled-components';

const StyledButton = styled.button(
  ({
    theme: {
      colors: { grey },
    },
    disabled,
    customStyles,
  }) => css`
    display: flex;
    padding: 0.5rem;
    background-color: transparent;
    border: 2px solid ${grey};

    ${disabled
      ? css`
          cursor: default;
          svg {
            fill: ${grey};
            transition: fill 0.2s;
          }
        `
      : css`
          & {
            background-color: transparent;
            transition: background-color 0.2s;
          }
          &:hover {
            background-color: rgba(0, 0, 0, 0.05);
            transition: background-color 0.2s;
          }
        `}

    ${customStyles}
  `,
);

const PaginationButton = ({ disabled, onClick, customStyles, children }) => {
  return (
    <StyledButton type="button" disabled={disabled} onClick={onClick} customStyles={customStyles}>
      {children}
    </StyledButton>
  );
};

export default PaginationButton;
