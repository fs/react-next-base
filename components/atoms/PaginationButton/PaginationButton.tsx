import React from 'react';
import type { PropsWithChildren } from 'react';

import { StyledButton } from './styled';
import ITest from 'types/test';

interface Props extends ITest {
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  disabled?: boolean;
  customStyles?: string;
}

const PaginationButton = ({ testId, disabled, onClick, customStyles, children }: PropsWithChildren<Props>): JSX.Element => {
  return (
    <StyledButton
      type="button"
      data-cy={testId}
      data-testid={testId}
      disabled={disabled}
      onClick={onClick}
      customStyles={customStyles}
    >
      {children}
    </StyledButton>
  );
};

export default PaginationButton;
