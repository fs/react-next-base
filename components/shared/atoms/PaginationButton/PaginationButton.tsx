import React from 'react';
import type { PropsWithChildren } from 'react';

import ITest from 'interfaces/testType';
import { StyledButton } from './styled';

interface Props extends ITest {
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  disabled?: boolean;
  customStyles?: string;
}

const PaginationButton = ({
  testId,
  disabled,
  onClick,
  customStyles,
  children,
}: PropsWithChildren<Props>): JSX.Element => {
  return (
    <StyledButton
      type="button"
      data-cy={testId}
      data-testid={testId}
      disabled={disabled}
      onClick={onClick}
      customStyles={customStyles}
    >
      {children}
    </StyledButton>
  );
};

export default PaginationButton;
