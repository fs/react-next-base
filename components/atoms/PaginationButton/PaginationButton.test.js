import React from 'react';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';

import renderWithTheme from '__tests__/helpers/renderWithTheme';

import PaginationButton from 'components/atoms/PaginationButton';

describe('PaginationButton', () => {
  const mockContent = "PaginationButton's content";

  test('should render correctly', () => {
    // Arrange

    // Act
    render(renderWithTheme(<PaginationButton>{mockContent}</PaginationButton>));

    const button = screen.getByRole('button');

    // Assert
    expect(button).toMatchSnapshot();
  });

  test('should has transparent background when disabled', () => {
    // Arrange

    // Act
    render(renderWithTheme(<PaginationButton disabled>{mockContent}</PaginationButton>));

    const button = screen.getByRole('button');

    // Assert
    expect(button).toHaveStyleRule('background-color', 'transparent');
  });
});
