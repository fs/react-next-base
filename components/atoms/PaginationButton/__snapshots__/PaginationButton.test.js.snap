// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationButton should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.5rem;
  background-color: transparent;
  border: 2px solid #ccc;
}

.c0 {
  background-color: transparent;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.c0:hover {
  background-color: rgba(0,0,0,0.05);
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

<button
  class="c0"
  type="button"
>
  PaginationButton's content
</button>
`;
